### PR TITLE
test-case: handle sof-logger errors in check-sof-logger.sh

### PR DESCRIPTION
**Pull request by aiChaoSONG**
Previously, we use the file size generated by sof-logger
to test if we have firmware log to export. But after a
logger change, sof-logger errors will output to generated
file, the size of the file is variant due to sof-logger
errors.
grep "errror|Error|ERROR“ from sof-logger's  stderr redirected file.

### CODE REVIEW

**Comment by aiChaoSONG** [comment]
sof-logger in sof repo seems need fix, because the output is ugly, let's wait that fix.

File: test-case/check-sof-logger.sh [hunk 1]
```diff
@@ -48,23 +48,29 @@ fi
 dlogi "Found file: $(md5sum $ldcFile|awk '{print $2, $1;}')"
 
 tmp_file=/tmp/$RANDOM.sof-logger.log
+err_tmp_file=/tmp/err_logger.log
```
Comment (xiulipan) [inline test-case/check-sof-logger.sh]: maybe we can also use the $RANDOM prefix to avoid duplicated issue?

File: test-case/check-sof-logger.sh [hunk 2]
```diff
@@ -48,23 +48,29 @@ fi
...
-dlogc "sudo $loggerBin -t -l $ldcFile -o $tmp_file 2>&1 &"
-sudo $loggerBin -t -l $ldcFile -o $tmp_file 2>&1 &
+# sof-logger errors will output to $err_tmp_file
+dlogc "sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &"
+sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &
 sleep 2
 dlogc "sudo pkill -9 $(basename $loggerBin)"
 sudo pkill -9 $(basename $loggerBin) 2> /dev/null
 
-# get size of trace log$
```
Comment (xiulipan) [inline test-case/check-sof-logger.sh]: can we still keep the trace size? There will also be issue that FW do not send trace to kernel as we have tried to checked in previous. We will need the Error check before this.

File: test-case/check-sof-logger.sh [hunk 3]
```diff
@@ -48,23 +48,29 @@ fi
...
+if [[ $logger_err ]]; then
+    dloge "No available log to export due to sof-logger errors."
+    sudo rm -rf $tmp_file $err_tmp_file
     exit 1
 fi
 dlogi "Log data BEG>>"
 cat $tmp_file
 dlogi "<<END log data"
+
+sudo rm -rf $tmp_file $err_tmp_file
```
Comment (xiulipan) [inline test-case/check-sof-logger.sh]: Do we need to clean up the tmp files? I think keep them for later debug is better.

File: test-case/check-sof-logger.sh [hunk 4]
```diff
@@ -48,23 +48,29 @@ fi
...
 
-# get size of trace log$
-size=`du -k $tmp_file | awk '{print $1}'`
-if [[ $size -lt 1 ]]; then
-    dloge "No available log export."
+# check if we get any sof-logger errors
+logger_err=`grep -E "Error|error|ERROR" $err_tmp_file`
+if [[ $logger_err ]]; then
+    dloge "No available log to export due to sof-logger errors."
+    sudo rm -rf $tmp_file $err_tmp_file
```
Comment (xiulipan) [inline test-case/check-sof-logger.sh]: same here.

File: test-case/check-sof-logger.sh [hunk 5]
```diff
@@ -48,23 +48,29 @@ fi
...
     exit 1
 fi
 dlogi "Log data BEG>>"
 cat $tmp_file
 dlogi "<<END log data"
+
+sudo rm -rf $tmp_file $err_tmp_file
+
```
Comment (Bin-QA) [inline test-case/check-sof-logger.sh]:
```suggestion
dlogi "Log data BEG>>"
cat $tmp_file
dlogi "<<END log data"
# check if we get any sof-logger errors
if [ "$(grep -i 'error' $tmp_file)" ]; then
    dloge "No available log to export due to sof-logger errors."
    sudo rm -rf $tmp_file
    exit 1
fi
sudo rm -rf $tmp_file 
```

Comment (aiChaoSONG) [reply test-case/check-sof-logger.sh]: let's first check sof-logger error, and then do log file size check. and as xiuli suggested, keep the log files for debug.

File: test-case/check-sof-logger.sh [hunk 6]
```diff
@@ -47,23 +47,34 @@ if [[ ! -f $ldcFile ]]; then
...
-sudo $loggerBin -t -l $ldcFile -o $tmp_file 2>&1 &
+# sof-logger errors will output to $err_tmp_file
+dlogc "sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &"
+sudo $loggerBin -t -l $ldcFile -o $tmp_file 2> $err_tmp_file &
 sleep 2
 dlogc "sudo pkill -9 $(basename $loggerBin)"
 sudo pkill -9 $(basename $loggerBin) 2> /dev/null
 
+# check if we get any sof-logger errors
+logger_err=`grep -i "error" $err_tmp_file`
```
Comment (Bin-QA) [inline test-case/check-sof-logger.sh]: if error record at `$tmp_file` ??

here 2 just for the stderr, but if 'error' dump into stdout

`logger_err=`grep -i "error"  $tmp_file $err_tmp_file``

Comment (xiulipan) [reply test-case/check-sof-logger.sh]: @Bin-QA @aiChaoSONG The logger update is to output error message to both stderr and stdout. So we need to only check stderr to check if error happen.

Comment (aiChaoSONG) [reply test-case/check-sof-logger.sh]: @Bin-QA @xiulipan if FW got error, the will be "error" keyword in the log,  checking both file, will break the logic.

Comment (Bin-QA) [reply test-case/check-sof-logger.sh]: if in both, why need `$err_tmp_file` if not in both, miss `$tmp_file` whether will lost something?

**Comment by aiChaoSONG** [comment]
@xiulipan Bin and I checked scripts that check FW and kernel, none of them checked FW log errors after boot. we may add this check in this script

File: test-case/check-sof-logger.sh [hunk 7]
```diff
@@ -47,23 +47,43 @@ if [[ ! -f $ldcFile ]]; then
...
+if [[ $logger_err ]]; then
+    dloge "No available log to export due to sof-logger errors."
+    exit 1
```
Comment (Bin-QA) [inline test-case/check-sof-logger.sh]: please add `$err_tmp_file` dump

Comment (aiChaoSONG) [reply test-case/check-sof-logger.sh]: Previously,  logger error info is duplicated in stdout and stderr, that why I don't dump here, but now, only the stderr.  This suggestion is good, and done and repushed

**Pull request merged**